pkg/registry/client.go

```diff
@@ -88,6 +88,45 @@ type (
 	ClientOption func(*Client)
 )
 
+// TODO remove in Helm 4
+// Deprecated
+// Fix for regression in v3.18.0
+// ORAS v1 was tolerant of an empty config file
+// ORAS v2 is not
+// to workaround this regression, ensure configfile is valid JSON
+func handleEmptyConfgFile(credentialsFile string, client *Client) {
+	f, err := os.Open(credentialsFile)
+	defer func(f *os.File) {
+		err := f.Close()
+		if err != nil {
+			client.err = err
+		}
+	}(f)
+	// handle only if credentials file exists
+	// ORAS already handles if file does not exist
+	if err == nil {
+		fileInfo, err := os.Stat(credentialsFile)
+		if err != nil {
+			client.err = err
+		}
+		// handle only if credentials file is empty
+		if fileInfo.Size() == 0 {
+			emptyMap := make(map[string]interface{})
+			jsonData, err := json.Marshal(emptyMap)
+			if err != nil {
+				client.err = err
+			}
+			// Attempt to write empty JSON map to config file
+			err = os.WriteFile(credentialsFile, jsonData, 0644)
```

Comment:
Two issues with this...

1. It makes a configuration file for credentials readable to those other than the user. This should be 0600.
2. If the file is mounted from a read only file system, this will have an error. That would not happen with v3.17. This can happen if Helm is run in a container.

Reply:
1. agree on mode, that's an easy fix.
2. passing a non-writable, empty credentials file feels like a far edge case, but technically agree it would have worked in 3.17.3 so fair enough. I like your bool approach in https://github.com/helm/helm/pull/30932 better. Could have applied that here but you already have it so closing this PR in favor of yours 👍 nice job there